code/game/machinery/vending/cm_vending.dm

```diff
@@ -1010,6 +1010,13 @@ GLOBAL_LIST_EMPTY(vending_products)
 		for(var/list/L as anything in box_list)
 			listed_products += list(L)
 
+///check if the item is actually in a tangible location. this will prevent duping via tactical reloads/attaching items and other means.
+/obj/structure/machinery/cm_vending/sorted/proc/check_item_loc(obj/item/item_to_stock, mob/user)
+	if(!is_type_in_list(item_to_stock.loc, list(/turf/, user, /obj/item/storage)))
```

Comment:
This is a really funky way to do this, and suggests you can simple put a storage in a storage to work around it. 

Also did you forget a distance check?

Regardless, why not use `get_turf` so that its always the turf that the thing is at?